docs/resources/core/instance.md

```diff
@@ -10,21 +10,38 @@ Gets a list of instances.
 ## Example Usage
 
 ```
-resource "oci_core_instance" "t" {
-    availability_domain = "availability_domain"
-    compartment_id = "compartment_id"
-    display_name = "display_name"
-    image = "imageid"
-    shape = "shapeid"
-    subnet_id = "subnetid"
-    metadata {
-        ssh_authorized_keys = "mypublickey"
-    }
-    extended_metadata {
-        some_string = "stringA"
-        nested_object = "{\"some_string\": \"stringB\", \"object\": {\"some_string\": \"stringC\"}}"
-    }
+resource "oci_core_instance" "testInstance" {
+	#Required
+	availability_domain = "${var.availability_domain}"
+	compartment_id = "${var.compartment_id}"
+	image = "${var.image_id}"
+	shape = "${var.shape}"
+
+	# Optional
+	create_vnic_details {
+		#Required (subnet_id may also be specified at the root level)
+		subnet_id = "${var.create_vnic_details_subnet_id}"
+
+		#Optional
+		assign_public_ip = "${var.create_vnic_details_assign_public_ip}"
+		display_name = "${var.create_vnic_details_display_name}"
+		hostname_label = "${var.create_vnic_details_hostname_label}"
+		private_ip = "${var.create_vnic_details_private_ip}"
+		skip_source_dest_check = "${var.skip_source_dest_check}"
+	}
+	display_name = "${var.display_name}"
+	hostname_label = "${var.hostname_label}"
```

Comment:
Good point.... will update, and also for subnet_id.

Ideally we should also mark these as deprecated in the schema (so that users get a warning/message when using it) and update all of our examples to not use it. I'll open a task to track that - I think it's fine to mark as deprecated in this doc without doing the rest for now - I don't want too much feature creep on this change.